## Aggregating ACS, FCC, Chicago Community Area Data

###### Importing Libraries

In [72]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

###### Importing Chicago FCC data

First, we use the FCC IL Dec 2020 csv file (which can be found [here](https://us-fcc.app.box.com/v/IL-Dec2020-v1)). Since it is a large file and we only want data on Chicago, we filter for Cook County data only and then export it as a new csv file which is saved in the "data" folder. **DO NOT RUN THE FOLLOWING CHUNKS OF CODE!!!**

In [ ]:
# FCC IL

fcc_df = pd.read_csv("data/IL-Fixed-Dec2020-v1.csv",
                     index_col=0,parse_dates=[0])

In [ ]:
#changing BlockCode column to string type

fcc_df['BlockCode']=fcc_df['BlockCode'].astype(str)

In [ ]:
# Extracting state, county, tract, block numbers from BlockCode column
# IL state=17
# Cook County=031

fcc_df['state'] = fcc_df['BlockCode'].str[:2]
fcc_df['county'] = fcc_df['BlockCode'].str[2:5]
fcc_df['tract'] = fcc_df['BlockCode'].str[5:11]
fcc_df['block'] = fcc_df['BlockCode'].str[-4:]

In [ ]:
# Filtering for Cook County only 
# 763788 rows

chi_fcc = fcc_df[(fcc_df.county == "031")]

In [ ]:
# dropping columns we don't need to make the file smaller

chi_fcc=chi_fcc[['ProviderName', 'Consumer', 
                 'MaxAdDown','MaxAdUp','tract']]
chi_fcc['tract']=chi_fcc['tract'].astype(float)

In [ ]:
# export final dataframe to csv file

chi_fcc.to_csv(r'data/chi_fcc.csv', index = False)

In [ ]:
# final dataframe looks like this
chi_fcc.head(5)

Run this code to load the Chicago FCC data.  

In [73]:
# Chicago FCC data

chi_fcc = pd.read_csv("data/chi_fcc.csv",index_col=0,parse_dates=[0])
chi_fcc.head(5)

Consumer  MaxAdDown  MaxAdUp    tract
ProviderName                                            
TOWERSTREAM, INC.         0        0.0      0.0  20702.0
TOWERSTREAM, INC.         0        0.0      0.0  80300.0
TOWERSTREAM, INC.         0        0.0      0.0  81000.0
TOWERSTREAM, INC.         0        0.0      0.0  81300.0
TOWERSTREAM, INC.         0        0.0      0.0  81500.0

###### Importing Chicago ACS aggregate and profile data

In [74]:
# ACS aggregate

acs_agg = pd.read_csv("data/acs5_aggregate.csv",index_col=0,
                      parse_dates=[0]).drop(['state', 'county'], axis=1)

In [75]:
# ACS profile

acs_pro = pd.read_csv("data/acs5_profile.csv",
                      index_col=0,
                      parse_dates=[0]).drop(
                        ['state', 'county','DP05_0001E'], axis=1)

In [76]:
# merging both ACS datasets
# renaming variables
# 1319 rows x 19 columns

acs_df = acs_agg.merge(acs_pro, on='tract').rename(columns={
                                'B01003_001E': 'total_pop',
                                'B28002_002E': 'has_internet',
                               'B28002_013E': 'no_internet',
                               'B28003_002E': 'has_computer',
                               'B28003_006E': 'no_computer',
                               'DP02_0001E': 'total_households',
                               'DP02_0152E': 'total_house_computer',
                               'DP02_0153E': 'has_broadband',
                               'DP03_0062E': 'median_income',
                               'DP03_0119PE': 'percent_poverty',
                               'DP05_0071E': 'total_hispanic',
                                'DP05_0078E': 'total_black'})

acs_df.head(5)

total_pop  has_internet  no_internet  has_computer  no_computer   tract  \
0       1825           392          149           426          149  630200   
1       5908          1242          231          1411          133  580700   
2       3419           928          140          1068          104  590600   
3       2835           917          138          1003           81  600700   
4       1639           322          245           356          218  611900   

   total_households  total_house_computer  has_broadband  median_income  \
0               575                   426            392          37422   
1              1544                  1411           1242          47000   
2              1172                  1068            917          46033   
3              1084                  1003            917          45294   
4               574                   356            322          24507   

   percent_poverty  total_hispanic  total_black  
0             25.7            1622            0  
1             17.4            4742          161  
2              7.9            2119            9  
3             17.0             850           82  
4             55.0             438         1175

###### Importing Chicago Community Area data

In [77]:
# Chi comm area
#rename columns

comm_area = pd.read_csv("data/chi_tracts.csv",
                        index_col=0,
                        parse_dates=[0]).rename(columns={
                        "Community Area Name": "name",          
                        "TRACT": "tract"})

In [78]:
# selecting columns we need 

comm_area=comm_area[['name', 'tract']]
comm_area.head(5)

name  tract
STUSAB                    
IL      Rogers Park  10100
IL      Rogers Park  10201
IL      Rogers Park  10202
IL      Rogers Park  10300
IL      Rogers Park  10400

Joining ACS data and Community Areas

In [79]:
# 803 rows x 20 columns

acs_comm = comm_area.merge(acs_df, on='tract')
acs_comm.head(5)

name  tract  total_pop  has_internet  no_internet  has_computer  \
0  Rogers Park  10100       4599          1801          460          2044   
1  Rogers Park  10201       7455          2091          434          2469   
2  Rogers Park  10202       2896           919          176          1022   
3  Rogers Park  10300       6485          2620          418          2883   
4  Rogers Park  10400       5213          1611          333          1774   

   no_computer  total_households  total_house_computer  has_broadband  \
0          319              2363                  2044           1789   
1          278              2747                  2469           2091   
2          114              1136                  1022            919   
3          208              3091                  2883           2620   
4          214              1988                  1774           1611   

   median_income  percent_poverty  total_hispanic  total_black  
0          32474             26.5             523         2045  
1          45639             28.6            1671         2481  
2          41486             16.4             753          974  
3          41250              6.6            1099         1995  
4          39700             13.9             392         1104

Joining FCC data and Community Areas

In [ ]:
# 372924 rows × 5 columns

fcc_comm = comm_area.merge(chi_fcc, on='tract')
fcc_comm.head(5)

###### Joining all datasets by tract

In [80]:
# merging all three dataframes by the column "tract"
# 372924 rows x 23 columns

full_df = comm_area.merge(acs_df,
                        on='tract').merge(chi_fcc,
                                          on='tract')


In [ ]:
full_df.head(5)

###### Computer, Internet, Broadband Access

This section will look into the community areas at a household-level to see who has and who does not have basic access to an internet and/or computer. We will also look into the households who have a broadband internet subscription. 

In [81]:
# who has a computer? who has internet access? 
# selecting columns we need

internet_df = acs_comm[['name', 
                        'total_households',
                        'no_internet', 
                        'has_internet', 
                        'has_computer', 
                        'no_computer',
                       'has_broadband']].groupby(by="name").sum()

In [82]:
# calculating who does not have a broadband internet subscription

internet_df['no_broadband']=internet_df['total_households']-internet_df['has_broadband']

In [83]:
# calculating percentages of households for each variable

# percentage of households with/out internet access

internet_df['no_internet']=internet_df['no_internet']/internet_df['total_households']*100
internet_df['has_internet']=internet_df['has_internet']/internet_df['total_households']*100

# percentage of households with/out no computer

internet_df['no_computer']=internet_df['no_computer']/internet_df['total_households']*100
internet_df['has_computer']=internet_df['has_computer']/internet_df['total_households']*100

# percentage of households with/out broadband 

internet_df['no_broadband']=internet_df['no_broadband']/internet_df['total_households']*100
internet_df['has_broadband']=internet_df['has_broadband']/internet_df['total_households']*100

In [84]:
# resulting dataframe

internet_df.sort_values(["has_broadband"],
                        ascending=True)

total_households  no_internet  has_internet  has_computer  \
name                                                                         
Burnside                      888    42.229730     53.716216     62.725225   
Englewood                    8983    34.598686     56.584660     74.897028   
West Englewood               9483    38.089212     57.239270     70.336391   
Fuller Park                  1128    34.663121     59.042553     69.414894   
North Lawndale              11075    29.986456     59.873589     76.000000   
...                           ...          ...           ...           ...   
Lincoln Square              18347     7.216439     89.960211     95.187224   
North Center                14093     8.238132     90.562691     95.217484   
Lincoln Park                32395     6.254052     91.106652     94.786232   
Lake View                   53480     5.731114     91.316380     96.394914   
Near South Side             14482     5.171938     91.175252     95.718823   

                 no_computer  has_broadband  no_broadband  
name                                                       
Burnside           37.274775      52.252252     47.747748  
Englewood          25.102972      56.306356     43.693644  
West Englewood     29.663609      57.239270     42.760730  
Fuller Park        30.585106      59.042553     40.957447  
North Lawndale     24.000000      59.620767     40.379233  
...                      ...            ...           ...  
Lincoln Square      4.812776      89.867553     10.132447  
North Center        4.782516      90.392393      9.607607  
Lincoln Park        5.213768      91.029480      8.970520  
Lake View           3.605086      91.071429      8.928571  
Near South Side     4.281177      91.175252      8.824748  

[77 rows x 7 columns]

Based on 2015-2019 ACS data, the percentages of households with computers, internet access, and internet broadband subscription are seen above. The neighborhoods of Burnside, Englewood, West Englewood, Fuller Park have the lowest percentages of both broadband subscription and internet access. The neighborhods of Near South Side, Lake View, Lincoln Park, and North Center have the highest. 

Broadband subscription and internet access numbers are extremely close, suggesting that the overwhelming majority of households who have access to internet do so via a broadbad subscription. 

###### Median Household Income & Poverty

In [85]:
# median household income and poverty rates by community area

income_df = acs_comm[['name', 
                      'median_income', 
                      'percent_poverty']].groupby(by = "name").median().sort_values(["percent_poverty"], 
                                                                                  ascending = False)

income_df

median_income  percent_poverty
name                                                  
Riverdale                     15408.0            49.20
Englewood                     23125.0            36.60
West Garfield Park            24001.5            36.40
East Garfield Park            26500.5            33.05
Greater Grand Crossing        29197.5            31.05
...                               ...              ...
Loop                         105094.0             2.30
Lincoln Park                 127177.5             1.85
North Center                 119904.5             1.80
Beverly                      111667.0             1.70
Lake View                    100816.0             1.50

[77 rows x 2 columns]

Median income and poverty rates are based on household-level data. The median household incomes in Chicago community areas range from Riverdale's \\$15,408 all the way to Lincoln Park's \\$127,177. Riverdale also has the highest percentage of their households in poverty at 49.2%. 

###### Race & Ethnicity

In [86]:
# race and ethnicity by community area

race_df = acs_comm[['name', 
                        'total_pop',
                        'total_hispanic', 
                        'total_black']].groupby(by = "name").sum().sort_values(["name"], 
                                                                      ascending = True)

In [87]:
# percentage of population hispanic

race_df['total_hispanic']=race_df['total_hispanic']/race_df['total_pop']*100

# percentage of population black non-hispanic

race_df['total_black']=race_df['total_black']/race_df['total_pop']*100

In [88]:
# final race and ethnicity dataframe

race_df.sort_values(["total_black"], ascending = False)

total_pop  total_hispanic  total_black
name                                                          
Avalon Park                  9713        0.092659    96.581901
Burnside                     2006        1.944167    96.261216
Washington Heights          26742        1.140528    96.096029
Chatham                     30967        0.846062    95.666355
Greater Grand Crossing      30149        1.738034    95.654914
...                           ...             ...          ...
Forest Glen                 19384       15.131036     1.191704
Jefferson Park              27503       24.746391     1.065338
Archer Heights,             13726       77.415125     0.954393
Norwood Park                43405       13.498445     0.799447
Edison Park                 11264       10.040838     0.372869

[77 rows x 3 columns]

Race and ethnicity data are based on total population numbers. Chicago community areas vary vastly in their race and ethnic compositions. Gage Park, South Lawndalw, West Elsdon, and Hermosa have the highest percentages of Hispanics/Latinos of all races. Calumet Heights, Washington Heights, Avalon Park, and Oakland have the highest percentages of non-Hispanic/Latino African-American/Blacks. 

##### Internet Access & Demographics Combined

The table below shows all of the variables above.

In [89]:
# merging all tables by community area

internet_demographics = internet_df.merge(income_df,
                        on='name').merge(race_df,
                                          on='name')

In [90]:
# reordering columns and showing final dataframe
# yellow indicates the maximum values in each column
# light blue indicates the minimum values in each column

internet_demographics[['total_pop',
                      'total_households',
                      'has_computer',
                      'has_internet',
                      'has_broadband',
                      'median_income',
                      'percent_poverty',
                     'total_hispanic',
                     'total_black']].sort_values(["name"], 
                                                 ascending = True).style.highlight_max(color='yellow').highlight_min(color = 'lightblue')

